examples/tests/private-https/Earthfile

```diff
@@ -0,0 +1,77 @@
+FROM ..+base
+
+all:
+    BUILD +git-clone-private-https-set-by-args
+    BUILD +git-clone-private-https-set-by-config-command
+    BUILD +git-clone-private-https-set-by-config-file
+    BUILD +git-clone-private-https-set-by-netrc
+
+
+git-clone-private-https-base:
```

Comment:
Maybe this target would be a cool UDC to share with the community?

Reply:
it's pretty hacky right now, but maybe we can get there eventually?

Maybe we could incorporate it into a blog post regarding these self-signed cert tests.